src/catalua.cpp

```diff
@@ -878,10 +885,10 @@ void CallbackArgument::Save()
             lua_pushstring( L, value_string.c_str() );
             break;
         case CallbackArgumentType::Tripoint:
-            LuaValue<tripoint>::push_reg( L, value_tripoint );
+            saved_registry_ref = LuaValue<tripoint>::push_reg( L, value_tripoint );
```

Comment:
Why is this stored in the registry in the first place? The value from the registry is never used and the values of other types above are not stored there.

Have you considered to *not* store it in the registry (change `push_reg` to `push`)?

Reply:
I don't know why it was written to store data in the registry in the first place. I assumed that was intentional behavior. If there are lua implementations with asynchronous garbage collection, it's possible the underlying memory could be deleted by lua after the lua callback but before the C++ side is done with the data; storing the data in the registry (temporarily) would be a way to prevent that by keeping a reference around, but the original author forgot to clean it up afterward.

If the original behavior of storing data in the registry was not intended, I can remove it as per your suggestion.

Reply:
In most of the other push calls here, the result is not stored in the registry either, why should it be needed for those two cases?

> If there are lua implementations with asynchronous garbage collection, it's possible the underlying memory could be deleted by lua after the lua callback but before the C++ side is done with the data; storing the data in the registry (temporarily) would be a way to prevent that by keeping a reference around, but the original author forgot to clean it up afterward.

That would be a bug in Lua. The pushed values must be there until the Lua callback (`lua_pcall`) returns (they are on the stack and therefor still references by Lua) and the C++ part of the code does not need them anyway. In any way: wouldn't that happen to the other values pushed here (but not stored in the registry)?

> If the original behavior of storing data in the registry was not intended

Seems like a sane assumption, at least it would be consistent.

Reply:
Changed so CallbackArgumentContainer::Save no longer stores data in the global lua registry, as per your suggestion.